src/components/messenger/list/index.tsx

```diff
@@ -171,34 +195,66 @@ export class Container extends React.Component<Properties, State> {
     );
   }
 
+  stringifyZero() {
+    const stringValue = this.props.zero.padStart(19, '0');
+    const whole = stringValue.slice(0, -18);
+    const decimal = stringValue.slice(-18).slice(0, 4).replace(/0+$/, '');
+    const decimalString = decimal.length > 0 ? `.${decimal}` : '';
+    return `${whole}${decimalString}`;
+  }
+
+  renderRewardsBar() {
+    return (
+      <div
+        className={classnames(c('rewards-bar'), {
+          [c('rewards-bar', 'with-avatar')]: this.props.includeRewardsAvatar,
+        })}
+      >
+        {this.props.includeRewardsAvatar && (
+          <Avatar size={'small'} type={'circle'} imageURL={this.props.userAvatarUrl} />
+        )}
+        <button
+          onClick={this.openRewards}
+          className={classnames(c('rewards-button'), {
+            [c('rewards-button', 'open')]: this.state.isRewardsPopupOpen,
+          })}
+        >
+          <div>Rewards</div>
+          <div className={c('rewards-icon')}>
+            <IconCurrencyDollar size={16} />
+            {this.props.includeRewardsAvatar && !this.state.isRewardsPopupOpenedOnce && (
+              <Status type='idle' className={c('rewards-icon__status')} />
+            )}
+          </div>
+        </button>
+      </div>
+    );
+  }
+
   render() {
     return (
       <>
         {this.props.includeTitleBar && this.renderTitleBar()}
-        <div
-          className={classnames(c('rewards-bar'), {
-            [c('rewards-bar', 'with-avatar')]: this.props.includeRewardsAvatar,
-          })}
-        >
-          {this.props.includeRewardsAvatar && (
-            <Avatar size={'small'} type={'circle'} imageURL={this.props.userAvatarUrl} />
-          )}
-          <button
-            onClick={this.openRewards}
-            className={classnames(c('rewards-button'), {
-              [c('rewards-button', 'open')]: this.state.isRewardsPopupOpen,
-            })}
+        {this.props.includeRewardsAvatar ? ( // only show the rewards tooltip popup if in full screen mode
```

Comment:
I think it'd be a good idea to use a separate flag here. You're assuming that the reason we want to show the rewards Avatar is the same reason we would want to show the popup. But, that may not always be the case. It's usually better to have the component render use very specific flags.

Reply:
done